lib/bunch/kv_enum.ex

```diff
@@ -0,0 +1,168 @@
+  def map_keys(enum, f) do
+    enum |> Enum.map(fn {key, value} -> {f.(key), value} end)
+  end
+
+  @doc """
+  Maps values of `enum` using function `f`.
+
+  ## Example
+
+      iex> #{inspect(__MODULE__)}.map_values([a: 1, b: 2], & &1+1)
+      [a: 2, b: 3]
+
+  """
+  @spec map_values(t(k, v1), (v1 -> v2)) :: t(k, v2) when k: any, v1: any, v2: any
+  def map_values(enum, f) do
+    enum |> Enum.map(fn {key, value} -> {key, f.(value)} end)
+  end
+
+  @doc """
+  Filters elements of `enum` by keys using function `f`.
+
+  ## Example
+
+      iex> #{inspect(__MODULE__)}.filter_by_keys([a: 1, b: 2, a: 3], & &1 == :a)
+      [a: 1, a: 3]
+
+  """
+  @spec filter_by_keys(t(k, v), (k -> as_boolean(term))) :: t(k, v) when k: any, v: any
+  def filter_by_keys(enum, f) do
+    enum |> Enum.filter(&apply_to_key(&1, f))
+  end
+
+  @doc """
+  Filters elements of `enum` by values using function `f`.
+
+  ## Example
+
+      iex> #{inspect(__MODULE__)}.filter_by_values([a: 1, b: 2, a: 3], & &1 |> rem(2) == 0)
+      [b: 2]
+
+  """
+  @spec filter_by_values(t(k, v), (v -> as_boolean(term))) :: t(k, v) when k: any, v: any
+  def filter_by_values(enum, f) do
+    enum |> Enum.filter(&apply_to_value(&1, f))
+  end
+
+  @doc """
+  Executes `f` for each key in `enum`.
+
+  ## Example
+
+      iex> #{inspect(__MODULE__)}.each_key([a: 1, b: 2, a: 3], & send(self(), &1))
+      iex> [:a, :b, :a] |> Enum.each(&receive do ^&1 -> :ok end)
+      :ok
+
+  """
+  @spec each_key(t(k, v), (k -> any | no_return)) :: :ok when k: any, v: any
+  def each_key(enum, f) do
+    enum |> Enum.each(&apply_to_key(&1, f))
+  end
+
+  @doc """
+  Executes `f` for each value in `enum`.
+
+  ## Example
+
+      iex> #{inspect(__MODULE__)}.each_value([a: 1, b: 2, a: 3], & send(self(), &1))
+      iex> 1..3 |> Enum.each(&receive do ^&1 -> :ok end)
+      :ok
+
+  """
+  @spec each_value(t(k, v), (v -> any | no_return)) :: :ok when k: any, v: any
+  def each_value(enum, f) do
+    enum |> Enum.each(&apply_to_value(&1, f))
+  end
+
+  @doc """
+  Returns `true` if `f` returns truthy value for any key from `enum`, otherwise `false`.
+
+  ## Example
+
+      iex> #{inspect(__MODULE__)}.any_key?([a: 1, b: 2, a: 3], & &1 == :b)
+      true
+      iex> #{inspect(__MODULE__)}.any_key?([a: 1, b: 3, a: 5], & &1 == :c)
+      false
+
+  """
+  @spec any_key?(t(k, v), (k -> as_boolean(term))) :: boolean when k: any, v: any
+  def any_key?(enum, f) do
+    enum |> Enum.any?(&apply_to_key(&1, f))
+  end
+
+  @doc """
+  Returns `true` if `f` returns truthy value for any value from `enum`, otherwise `false`.
+
+  ## Example
+
+      iex> #{inspect(__MODULE__)}.any_value?([a: 1, b: 2, a: 3], & &1 |> rem(2) == 0)
+      true
+      iex> #{inspect(__MODULE__)}.any_value?([a: 1, b: 3, a: 5], & &1 |> rem(2) == 0)
+      false
+
+  """
+  @spec any_value?(t(k, v), (v -> as_boolean(term))) :: boolean when k: any, v: any
+  def any_value?(enum, f) do
+    enum |> Enum.any?(&apply_to_value(&1, f))
+  end
```

Comment:
I think `any_key?/1` and `any_value?/1` variants may be handy that would follow Elixir's `if/2` rules in place of custom function `f`:

```elixir
iex(1)> if 2, do: :oke
:oke
iex(2)> if nil, do: :oke
nil
iex(3)> if false, do: :oke
nil
iex(4)> if true, do: :oke 
:oke
```

Reply:
oke majfrend